core/include/kernel/callout.h

```diff
@@ -0,0 +1,112 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2024, Linaro Limited
+ */
+
+#ifndef __KERNEL_CALLOUT_H
+#define __KERNEL_CALLOUT_H
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <sys/queue.h>
+
+/*
+ * struct callout - callout reference
+ * @callback:	function to be called when a callout expires, internal use only
+ * @ctrval:	callout expiry time counter value, internal use only
```

Comment:
Using milliseconds has the drawback that we may need to divide the time counter value with the frequency quite often. I suppose it would mean to implement `timer_now()` as
`return barrier_read_counter_timer() * 1000 / read_cntfrq()`.

Will `expiry_value` work better?